flyteadmin/auth/authzserver/metadata_provider.go

```diff
@@ -107,3 +121,45 @@ func NewService(config *authConfig.Config) OAuth2MetadataProvider {
 		cfg: config,
 	}
 }
+
+func sendAndRetryHttpRequest(ctx context.Context, client *http.Client, url string, retryAttempts int, retryDelay time.Duration) (*http.Response, error) {
+	var response *http.Response
+	var err error
+	totalAttempts := retryAttempts + 1 // Add 1 for the case where 0 retryAttempts are specified
```

Comment:
can we update this comment? we always add 1 regardless of whether retryAttempts == 0

Reply:
Let me know if it sounds better to you!